File: lethe/tools/neat/neat.go
// Package neat implements value sanitisation functions.
package neat

import (
	"path/filepath"
	"strings"
	"unicode"
)

// Body returns a trimmed file body string with a trailing newline.
func Body(body string) string {
	return strings.TrimSpace(body) + "\n"
}

// Extn returns a trimmed lowercase file extension string with a leading dot.
func Extn(extn string) string {
	extn = strings.ToLower(extn)
	extn = strings.TrimSpace(extn)
	return "." + strings.TrimPrefix(extn, ".")
}

// Name returns a trimmed lowercase alphanumeric file name string.
func Name(name string) string {
	var chars []rune
	for _, char := range strings.ToLower(name) {
		switch {
		case unicode.IsLetter(char), unicode.IsNumber(char):
			chars = append(chars, char)
		case unicode.IsSpace(char), char == '-', char == '_':
			chars = append(chars, '-')
		}
	}

	return strings.Trim(string(chars), "-")
}

// Path returns a trimmed normalised file path string.
func Path(path string) string {
	path = strings.TrimSpace(path)
	return filepath.Clean(path)
}

File: lethe/tools/neat.py
"""
The 'lethe.tools.neat' module implements value sanitisation and conversion functions.
"""

import os.path
import pathlib
import string

NAME_CHARS = string.ascii_letters + string.digits + "-_"


def body(body: str) -> str:
    """
    Return a whitespace-trimmed file body string with a trailing newline.
    """

    return body.strip() + "\n"


def glob(glob: str) -> str:
    """
    Return a whitespace-trimmed lowercase glob pattern string.
    """

    return glob.strip().lower()


def name(name: str) -> str:
    """
    Return a whitespace-trimmed lowercase alphanumeric file name string.
    """

    name = "".join(char for char in name.replace(" ", "-") if char in NAME_CHARS)
    return name.strip("-").lower()


def path(path: str | pathlib.Path) -> str:
    """
    Return a whitespace-trimmed normalised file path string.
    """

    path = str(path).strip()
    return os.path.normpath(path)
